js/packages/openinference-instrumentation-llama-index/src/utils.ts

```diff
@@ -56,14 +63,94 @@ function getExecContext(span: Span) {
   return execContext;
 }
 
+/**
+ * If execution results in an error, push the error to the span.
+ * @param span
+ * @param error
+ */
+function handleError(span: Span, error: Error | undefined) {
+  if (error) {
+    span.recordException(error);
+    span.setStatus({
+      code: SpanStatusCode.ERROR,
+      message: error.message,
+    });
+    span.end();
+  }
+}
+
+function documentAttributes(
+  output: llamaindex.NodeWithScore<llamaindex.Metadata>[],
+) {
+  const docs: Attributes = {};
+  output.forEach(({ node, score }, idx) => {
+    if (node instanceof llamaindex.TextNode) {
+      const prefix = `${SemanticConventions.RETRIEVAL_DOCUMENTS}.${idx}`;
+      docs[`${prefix}.${SemanticConventions.DOCUMENT_ID}`] = node.id_;
+      docs[`${prefix}.${SemanticConventions.DOCUMENT_SCORE}`] = score;
+      docs[`${prefix}.${SemanticConventions.DOCUMENT_CONTENT}`] =
+        node.getContent();
+      docs[`${prefix}.${SemanticConventions.DOCUMENT_METADATA}`] =
+        safelyJSONStringify(node.metadata) ?? undefined;
+    }
+  });
+  return docs;
+}
+
+function queryEmbeddingAttributes(input: string, output: number[]) {
+  return {
+    [`${SemanticConventions.EMBEDDING_EMBEDDINGS}.0.${SemanticConventions.EMBEDDING_TEXT}`]:
+      input,
+    [`${SemanticConventions.EMBEDDING_EMBEDDINGS}.0.${SemanticConventions.EMBEDDING_VECTOR}`]:
+      output,
+  };
+}
+
+function textEmbeddingsAttributes(input: string[], output: number[][]) {
```

Comment:
same

Reply:
see https://github.com/Arize-ai/openinference/pull/761/files#r1706074625 comment but you don't need this an d the function above, i would just use this one and in the case of one embedding just wrap it in a list when you pass it in